include/avif/avif.h

```diff
@@ -7,23 +7,39 @@
 #include <stddef.h>
 #include <stdint.h>
 
-#ifndef AVIF_API
-#if defined(AVIF_BUILDING_SHARED_LIBS)
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// ---------------------------------------------------------------------------
+// Export macros
+
+// AVIF_BUILDING_SHARED_LIBS should only be defined when libavif is being built
+// as a shared library.
+// AVIF_DLL should be defined if libavif is a shared library. If you are using
+// libavif as CMake dependency, through CMake package config file or
+// through pkg-config, this is defined automatically.
```

Comment:
Optional: I like your summary in the commit message very much:

> Here's what `AVIF_API` will be defined to:
> | (Shared) | Windows | Unix |
> | --- | --- | --- |
> | Build | `__declspec(dllexport)` | `__attribute__((visibility("default")))` |
> | Use | `__declspec(dllimport)`| |
> 
> For static build, `AVIF_API` always defines to nothing. 

Do you think a plaintext versiopn of that (without the table markup) could become a comment here? Or do you think that comment would merely repeat the code?

Reply:
I think it can help the future maintenance of the project by explaining what's going on here cleanly. I would like to add it.